dataland-frontend/src/App.vue

```diff
@@ -1,68 +1,131 @@
+      return useSharedSessionStateStore().refreshToken;
+    },
+  },
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  async created() {
+    this.keycloakPromise = this.initKeycloak();
+    this.resolvedKeycloakPromise = await this.keycloakPromise;
+    if (this.resolvedKeycloakPromise && this.resolvedKeycloakPromise.authenticated) {
+      updateTokenAndItsExpiryTimestampAndStoreBoth(this.resolvedKeycloakPromise, true);
+    }
+  },
+
+  provide() {
+    return {
+      getKeycloakPromise: (): Promise<Keycloak> => {
+        if (this.keycloakPromise) return this.keycloakPromise;
+        throw new Error("The Keycloak promise has not yet been initialised. This should not be possible...");
+      },
+      authenticated: computed(() => {
+        return this.keycloakAuthenticated;
+      }),
+    };
+  },
+
   methods: {
     /**
-     * Sets up the keycloak and keycloakPromise objects that are passed down
-     * to the other components to handle authentication
+     * Initializes the Keycloak adaptor and configures it according to the requirements of the Dataland application.
+     *
+     * @returns a promise which resolves to the Keycloak adaptor object
      */
-    initKeycloak() {
-      const initOptions = {
-        realm: "datalandsecurity",
-        url: "/keycloak",
-        clientId: "dataland-public",
-        onLoad: "login-required",
-      };
-      const keycloak = new Keycloak(initOptions);
-      const keycloakPromise = keycloak
+    initKeycloak(): Promise<Keycloak> {
+      const keycloak = new Keycloak(KEYCLOAK_INIT_OPTIONS);
+      const handleAuthLogoutBound = this.handleAuthLogout.bind(this);
+      keycloak.onAuthLogout = handleAuthLogoutBound;
+      return keycloak
         .init({
           onLoad: "check-sso",
           silentCheckSsoRedirectUri: window.location.origin + "/silent-check-sso.html",
           pkceMethod: "S256",
         })
-        .then((authenticated): boolean => {
+        .then((authenticated) => {
           this.keycloakAuthenticated = authenticated;
-          return authenticated;
         })
         .catch((error) => {
-          console.log("Error in init keycloak ", error);
+          console.log("Error in init Keycloak ", error);
           this.keycloakAuthenticated = false;
         })
         .then((): Keycloak => {
           return keycloak;
         });
+    },
 
-      this.keycloak = keycloak;
-      this.keycloakPromise = keycloakPromise;
+    /**
+     * Executed as callback when the user is logged out. It tries another logout by redirecting the user to a keycloak
+     * logout uri, where the user is instantly re-redirected back to the Welcome page with a specific query param
+     * in the url which triggers a pop-up to open and inform the user that she/he was just logged out.
+     */
+    handleAuthLogout() {
+      logoutAndRedirectToUri(this.resolvedKeycloakPromise as Keycloak, "?externalLogout=true");
```

Comment:
Warum muss hier nochmal ausgeloggt werden? Wird die Funktion nicht nur dann aufgerufen, wenn ein Logout stattgefunden hat?

Reply:
Wenn ich mich recht erinnere, dann hat ein purer "redirect" hier nicht funktioniert.
Philip hatte dann vorgeschlagen, einfach nochmal den "logout"+redirect zu machen, auch wenn das logout eigentlich redundant ist.

Ich kann nochmal ausprobieren, ob ich es mit einem puren redirect hinbekomme und dann hier schreiben, wie es lief.

Reply:
Outcome: es wäre konsistenter wenn man die external logout warnung auf der Seite sieht wo man gerade ist, und die einzigen Optionen sind 
1.) Tab schließen
2.) Login

Wenn sehr aufwendig, dann lass es.

Reply:
Habe es ausprobiert.  Mein Fazit ist, dass sich der Aufwand nicht lohnt:

1.) Wenn das Popup auf der Seite auftaucht, auf der man aktuell ist, eröffnet das einige "Freiheitsgrade",  wodurch einige neue Bugs/komische Siuationen erstmalig aufgetreten sind.
Bestimmt kann man das alles mit etwas Aufwand abfangen. Aber:
Es ist insgesamt einfach simpler, pauschal auf die Startseite weiterzuleiten und dort das Popup zu zeigen.

2.) Nach etwas Überlegen bin ich für mich zu dem Schluss gekommen, dass ich es sogar (leicht) präferieren würde, dass man auf die Welcome Page geleitet und dann dort das popup angezeigt bekommt.
Grund:  Wenn du rausgeschmissen wirst, weil du dich in einem anderen Tab ausloggst, dann kann man auch davon ausgehen dass du für dich beschlossen hast, dass du "fertig" mit deiner Arbeit auf Dataland bist.
Daher ist es doch auch ok, wenn du auf der Startseite landest.
Du dürftest dich ja in keinem FAll darüber "wundern".

Was meinst du?